zookeeper-api/src/test/java/org/apache/helix/zookeeper/impl/client/RealmAwareZkClientTestBase.java

```diff
@@ -67,4 +76,127 @@ public void afterClass() {
       _msdsServer.stopServer();
     }
   }
-}
+  /**
+   * Initialize requirement for testing multi support.
+   */
+  @Test
+  public void testMultiSetup() {
+    // Create a connection config with a valid sharding key
+    RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder builder =
+            new RealmAwareZkClient.RealmAwareZkConnectionConfig.Builder();
+    RealmAwareZkClient.RealmAwareZkConnectionConfig connectionConfig =
+            builder.setZkRealmShardingKey(ZK_SHARDING_KEY_PREFIX).build();
+    try {
+      _realmAwareZkClient = new FederatedZkClient(connectionConfig,
+              new RealmAwareZkClient.RealmAwareZkClientConfig());
+    } catch (IllegalArgumentException e) {
+      Assert.fail("Invalid Sharding Key.");
+    } catch (Exception e) {
```

Comment:
can we integrate these 2 catches?

Reply:
how would you integrate both? Each catch is to specify a different error origin

Reply:
Just to share a different thought, if the exception is abnormal, maybe we don't need to catch and check at all. Just let it throw.

Reply:
If you want to integrate these two, then you can just catch `Exception ` since `IllegalArgumentException` is a subclass of general `Exception ` ( https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html)
Throw the unexpected exception also work as the stack trace will contain enough info. 
In my opinion both work.

Reply:
Sound good, I think think I'll have the exceptions be thrown without catching them then. Thank you for the input!